features/integration/applications.feature

```diff
@@ -125,15 +126,82 @@ Feature: Applications
+      And I sleep for 500 milliseconds for indexer to digest things down.
+      And according to "indexer", the contents of the box with name "str:foo bar" in the current application should be "YmF6IHF1eAAAAAAAAAAAAAAAAAAAAAAA". If there is an error it is "".
+
+      # full check confirmed by both algod and indexer
+      Then according to "algod", the current application should have the following boxes "Zm9vIGJhcg==:APj/IA==:bmFtZQ==:MTE0NTE0".
+      And according to "algod", with 2 being the parameter that limits results, the current application should have 2 boxes.
+      And according to "algod", with 6 being the parameter that limits results, the current application should have 4 boxes.
+
+      And I sleep for 500 milliseconds for indexer to digest things down.
+      And according to "indexer", the current application should have the following boxes "Zm9vIGJhcg==:APj/IA==:bmFtZQ==:MTE0NTE0".
+      And according to "indexer", with 2 being the parameter that limits results, the current application should have 2 boxes.
+      And according to "indexer", with 6 being the parameter that limits results, the current application should have 4 boxes.
+
+      # delete one box
+      Then I build an application transaction with the transient account, the current application, suggested params, operation "call", approval-program "", clear-program "", global-bytes 0, global-ints 0, local-bytes 0, local-ints 0, app-args "str:delete,str:name", foreign-apps "", foreign-assets "", app-accounts "", extra-pages 0, boxes "0,str:name"
+      And I sign and submit the transaction, saving the txid. If there is an error it is "".
+      And I wait for the transaction to be confirmed.
+      And according to "algod", the current application should have the following boxes "Zm9vIGJhcg==:APj/IA==:MTE0NTE0".
+
+      # move to indexer testing steps
+      And I sleep for 500 milliseconds for indexer to digest things down.
+      And according to "indexer", the contents of the box with name "str:name" in the current application should be "". If there is an error it is "no application boxes found".
+      And according to "indexer", the contents of the box with name "b64:APj/IA==" in the current application should be "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA". If there is an error it is "".
+      And according to "indexer", the current application should have the following boxes "Zm9vIGJhcg==:APj/IA==:MTE0NTE0".
+      And according to "indexer", with 2 being the parameter that limits results, the current application should have 2 boxes.
+      And according to "indexer", with 0 being the parameter that limits results, the current application should have 3 boxes.
+      And according to indexer, with 1 being the parameter that limits results, and "b64:APj/IA==" being the parameter that sets the next result, the current application should have the following boxes "MTE0NTE0".
```

Comment:
Perhaps we can roll this step up with: `And according to <client>, the current application should have the following boxes "Zm9vIGJhcg==:APj/IA==:MTE0NTE0".`:

We could combine the two and have a step like:
`And according to <client>, with limit <max> and next "<next>", the current application should have the following boxes "<boxes>".` and ignore `next` if it is an `algod` client.

Reply:
This is also somewhat brought up by Zeph previously in a thread of discussion: https://github.com/algorand/algorand-sdk-testing/pull/232#discussion_r969017995, and my understanding on endpoint *boxes* result returning is that, if we ask with `limit`, we are asking for a subset of all boxes under an app.

A problem directly arise: which part of subset is the endpoint returning, or how can we manipulate the order such that we can get the result for expected boxes? These are implicit and hidden. Without special notifying, the test steps in scenario cannot be easily swapped, which smells error prone and hard to maintain for me.

Reply:
(I mean, the `limit` case only applies on algod, for indexer it seems returning is ordered? but this is still an obstacle)

Reply:
yea that makes sense, I can drop a note in testcase to explain the thinking process.

Reply:
thanks! comment looks good so resolving